src/portable/project.rs

```diff
@@ -192,6 +192,7 @@ pub struct Upgrade {
 pub struct Handle {
     name: String,
     instance: InstanceKind,
+    project_dir: Option<PathBuf>,
```

Comment:
I think `None` here inconsistently means either "no set/unknown" or "current dir". We probably should add `project_dir` argument to `Handle::probe`. And then probably do not make the field optional (if possible).